### PR TITLE
Fix #7416: Don't emit signatures of fields of primitive types

### LINKED ISSUE #7416: Bad signature for private[this] field of value class type
How to reproduce:

In Line 26 of transform/CapturedVars.scala, change

    private var Captured: Store.Location[collection.Set[Symbol]] = _

to

    private[this] var Captured: Store.Location[collection.Set[Symbol]] = _

Compiling CapturedVars with -Xverify-signature gives

```
checking CapturedVars.scala after phase genSJSIR
-- Error: CapturedVars.scala:26:14 ---------------------------------------------
26 |  private var Captured: Store.Location[collection.Set[Symbol]] = _
   |              ^
   |compiler bug: created invalid generic signature for variable Captured in dotty.tools.dotc.transform.CapturedVars
   |signature: I
   |if this is reproducible, please report bug at https://github.com/lampepfl/dotty/issues
Error while emitting CapturedVars.scala
exception occurred while compiling CapturedVars.scala
java.lang.IllegalArgumentException: I: 'T' expected at index 0 while compiling CapturedVars.scala
Exception in thread "main" java.lang.IllegalArgumentException: I: 'T' expected at index 0
	at scala.tools.asm.util.CheckClassAdapter.checkChar(CheckClassAdapter.java:1018)
	at scala.tools.asm.util.CheckClassAdapter.checkTypeVariableSignature(CheckClassAdapter.java:951)
	at scala.tools.asm.util.CheckClassAdapter.checkFieldTypeSignature(CheckClassAdapter.java:859)
	at scala.tools.asm.util.CheckClassAdapter.checkFieldSignature(CheckClassAdapter.java:714)
	at dotty.tools.backend.jvm.DottyBackendInterface.verifySignature$$anonfun$1(DottyBackendInterface.scala:496)
	at dotty.runtime.function.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.backend.jvm.DottyBackendInterface.wrap$1(DottyBackendInterface.scala:480)
	at dotty.tools.backend.jvm.DottyBackendInterface.verifySignature(DottyBackendInterface.scala:501)
```

It seems that the signature checker does not accept fields of primitive types (presumably, since a generic signature for such fields would be useless). In this case, the signature 
comes from `Store.Location` which is a value class wrapping an `Int`. 

If `Captured` is just `private` but not `private[this]`, the signature is generated for the getter instead.

### PR DESCRIPTION
**Pull request by odersky**
Don't emit signatures for private fields of primitive type.

Maybe there's a better fix for this?

### CODE REVIEW

**Comment by smarter** [comment]
scalac has a very similar fix: https://github.com/scala/scala/pull/5925/commits/e3bee067121f3a3f1d78bde8e04ebe77d7000a3a

File: compiler/src/dotty/tools/dotc/core/TypeErasure.scala [hunk 1]
```diff
@@ -147,6 +147,12 @@ object TypeErasure {
   def valueErasure(tp: Type)(implicit ctx: Context): Type =
     erasureFn(isJava = false, semiEraseVCs = true, isConstructor = false, wildcardOK = false)(tp)(erasureCtx)
 
+  /** Like value class erasure, but value classes erase to their underlying type erasure */
+  def fullErasure(tp: Type)(implicit ctx: Context): Type =
+    valueErasure(tp) match
```
Comment (Duhemm) [inline compiler/src/dotty/tools/dotc/core/TypeErasure.scala]: Was there some decision about mixing significant indentation syntax with braces syntax? (this `match` has no braces)

Comment (odersky) [reply compiler/src/dotty/tools/dotc/core/TypeErasure.scala]: I currently use indentation syntax for all code I touch to get a better feeling for it. If we decide not to do indentation, we can rewrite to braces automatically. 

The reason not to convert to indentation wholesale is that is would destroy commit history.

**Comment by odersky** [comment]
I don't have time to port the test (we do not have an analogue of GenericSignaturesTest, it seems) but if someone could do it that would be welcome!

**Pull request merged**